data-transform/src/main/java/com/linkedin/data/transform/filter/AbstractFilter.java

```diff
@@ -248,21 +260,33 @@ private Object filterDataMap(DataMap opNode,
     for (Map.Entry<String, Object> entry : valueDataMap.entrySet())
     {
       final String name = entry.getKey();
+      final String nameEscaped = Escaper.replaceAll(name, "$", "$$");
```

Comment:
If the name is already escaped(replaced "$" to "$$"), could it happen?

Reply:
Field names in the data are never escaped. We only escape them in the mask tree as it has special meaning for $ (eg $*, $start etc)